### PR TITLE
Removed space before ellipses in German translation

### PR DESCRIPTION
**Pull request by dg9ngf**
These extra spaces are one of the first things I noticed after installing Notepad++. They're not used anywhere else so I removed them. No other changes.

### CODE REVIEW

**Comment by dg9ngf** [comment]
There are actually lots of other inconsistencies and errors in the German translation (inconsistent wording, multiple hotkey definition) and I'm currently correcting that file. Expect an updated pull request soon.

**Comment by h-h-h-h** [comment]
**I am against the removal of spaces before ellipses. It is common for German software translations to have a space before ellipses, e.g., in menu items or buttons.**

As a reference, you can take a look at [GIMP's](http://www.gimp.org/) German menu items, e.g.

**Comment by dg9ngf** [comment]
I have indeed found a Wikipedia article that shows using a space before an ellipsis for GUI command texts *in a single example*, but there's no explanation or reference for that given nor is the use of a space explicitly mentioned here, so it's unverifyable and might be a mistake. Let me list other applications that don't use a space before the ellipsis:

* Every single application part of the Windows operating system
* Adobe Lightroom
* Adobe Photoshop
* Avira AntiVirus
* Beyond Compare
* CDBurnerXP
* DVBViewer
* Google Earth
* InkScape
* IrfanView
* Microsoft Office
* Microsoft Visual Studio
* Mozilla Firefox
* Mozilla Thunderbird
* Paint.net
* Pidgin
* Speed Commander
* TeamViewer
* TextPad

Do you know any of them? Some of these applications were made in Germany, others are professionally translated. I did not find any application on my computer with a German UI that uses a space in that place, except Notepad++. Now you say all of them are wrong? Prove it.

As for where ellipses are used at all, I find you link an interesting reference and will look through all menu commands again to correct the use of "…" where necessary.

**Comment by xylographe** [comment]
@dg9ngf While you're at it, add a translation for `Count: X matches.` in the **Find** dialogue, *please*.

**Comment by h-h-h-h** [comment]
> @dg9ngf:
there's no explanation or reference for that given nor is the use of a space explicitly mentioned here, so it's unverifyable and might be a mistake.

I think Wikipedia would not be *the* reference, anyhow&mdash;it would be just *one* reference. But you're right&mdash;webpages discussing this topic are extremely hard to find. Unfortunately, I couldn't find any (either positive or negative).

> I did not find any application on my computer with a German UI that uses a space in that place, except Notepad++. Now you say all of them are wrong? Prove it.

In English, the use of a space before an ellipsis is handled differently ([reference](http://english.stackexchange.com/questions/26240/what-is-the-proper-way-of-using-triple-dots-and-spaces-before-after-them)). In German, no space means omitting *characters*, and a space means omitting *words*. <q>Omitting a dialog</q> should be represented as omitting a word.

In my opinion not using a space before ellipses is a somehow naive transfer of what is common *in English*. It's a pity for me that even Windows programs as Notepad don't use a space before ellipses in menu items. However, there's a number of software (as open-source software) that adheres to using a space before ellipses:
- Audacity (a few inconsistent items remained)
- GIMP
- VLC media player

In XnViewMP one menu item reads: <q>TWAIN-Quelle wählen (Scanner, Kamera, ...)...</q> When you compare this with [a Wikipedia article](https://de.wikipedia.org/wiki/Auslassungspunkte#Aufz.C3.A4hlung), you see that people mostly don't adhere to a thought-out concept: <q>Er beherrscht alle lateinamerikanischen Tänze: Tango, Salsa, Rumba …</q> Pay attention to the comma and to the style of writing <q>...)...</q>

---

> @xylographe:
While you're at it, add a translation for Count: X matches. in the Find dialogue, please.

What's also missing is a translation for:
- Failed matches
- Beginning to match from the top again
- Replacement information (in either case&mdash;one or all)

**Comment by xylographe** [comment]
In the context of menu items ellipses do not indicate an omission:

    Section *Using ellipses* in [*Menus* at **MSDN**](https://msdn.microsoft.com/en-us/library/windows/desktop/dn742392.aspx)

**Comment by dg9ngf** [comment]
Correct, nothing is omitted here so these rules do not apply.

Does anybody know how ellipses at the end of sentences (not as omission) are correctly written in English? I couldn't find consistent information in Wikipedia. If they expect a space as well (like in German), then menu labels seem to define their own category and their use cannot be inferred from text writing rules. If that's the case, I'd strongly suggest doing it the way everybody does it and not stand out.

(And even if not, I'd question why we should not follow common practice. Microsoft and Adobe do have some weight in this matter. Think of "Kennwort" when people commonly use the [Denglish](http://dictionary.reference.com/browse/denglish) word "Passwort" because it's "password" in English. Open-source software is sadly often translated by people who lack the necessary sense of detail which can sometimes be seen on the user interface as well. Large commercial software makers employ specially trained experts for these things.)

**Comment by h-h-h-h** [comment]
[The Wikipedia article](https://de.wikipedia.org/wiki/Auslassungspunkte#Aufz.C3.A4hlung) acknowledges that an ellipsis doesn't necessarily indicate an omission (see also the following captions). Yet, an ellipsis without a space is not used these cases. It may be viewed as <q>a kind of word omission,</q> maybe a <q>content omission,</q> in these cases.

> @dg9ngf:
Large commercial software makers employ specially trained experts for these things.

The MSDN documentation pages seem to be automatically (or sometimes semi-automatically) translated. So it is questionable that they definitely payed attention to this detail for the German language.

Also, <q>&nbsp;-&nbsp;</q> is still used regularly, although a proper n-dash (<q>&nbsp;&ndash;&nbsp;</q>) would not be a problem with fonts like Segoe UI and windows being Unicode-enabled (reference: Notepad; see also [this Wikipedia article](https://de.wikipedia.org/wiki/Halbgeviertstrich#Gedankenstrich)). I think there's no good reason for Windows to ignore these typographical rules.

> Microsoft and Adobe do have some weight in this matter.

Programs as VLC media player are renowned software that could similarly be taken as a reference.

> Does anybody know how ellipses at the end of sentences (not as omission) are correctly written in English?

It seems spaces around ellipses are recommended, but not the only common way.

**Comment by dg9ngf** [comment]
> The Wikipedia article acknowledges that an ellipsis doesn't necessarily indicate an omission

But the menu command isn't an enumeration either. Still doesn't apply.

> The MSDN documentation pages seem to be automatically

Only recently. It wasn't the case in earlier versions. Even today, "automatically translated" doesn't mean it threw away what was manually translated before and hasn't changed. Unfortunately, they sometimes forgot to update localised sites so they are partly outdated and misleading. And they really messed it up with their translation help nagging. That's why I actually [prefer the en-US site](http://unclassified.software/source/msdn-english) today. Not for the language of manually translated content.

> Also, " - " is still used regularly, although a proper en dash (" – ") would not be a problem with fonts like Segoe UI

But it is a problem with the commonly used keyboard layout. Do you always want to press Alt+0150 when you have a hyphen at hand? Blame it for keyboard layouts. The same applies to quotation marks or even the ellipsis. I've made my own keyboard layout so I can use them, but still Shift+2 is easier than what was still free for more advanced characters (AltGr+Shift+2 and AltGr+Shift+3 in my case).

> Programs as VLC media player are renowned software

For technical implementation, yes. Certainly not for UI. Linux is a very good examply IMO. It's been the superior server platform from the start but couldn't catch up on the desktop for a very long time. Today, both are learning from each other, finally.

Any more arguments? Can we get back to business?

I'll look into the other issues mentioned above but I likely won't get to it until next week.

**Comment by h-h-h-h** [comment]
> @dg9ngf:
Any more arguments?

Sure! I'm still against it, and I would like to see some more votes. I think @janschreiber translated many items. Let's see whether he hears that.

> the menu command isn't an enumeration either.

The Wikipedia article talks about many applications for an ellipsis, and the cases with surrounding spaces are the vast majority. All these cases have to do with a more generic point of view towards omitting characters. For me it's clear which category (spaces or not) menu items with an ellipsis belong to. In a sense, these two examples are applicable in my opinion:

- Stefan begann zu stottern: „Ich – ich weiß nicht …“
- Das ist der Stoff, aus dem die Träume sind …

> But it is a problem with the commonly used keyboard layout. Do you always want to press Alt+0150 when you have a hyphen at hand? Blame it for keyboard layouts. The same applies to quotation marks or even the ellipsis.

I'm talking about the development. In Notepad, e.g., the title should be hardcoded like this: <q>%s &ndash; Editor</q>. It's not a big deal to go through the strings of an application and replace some simple substitution characters (as `""` instead of <code><q></q></code>) with the typographically correct ones (as <code><q></q></code>). If you're concerned about inconsistency of UI and user-typed data: I think, the UI shouldn't reflect the simplicity of user-typed data with substitution characters. The user understands, that *he/she* typed that data and is <q>responsible</q> for that. Some application might even automatically replace the characters, as Word does it with quotation marks.

**Comment by dg9ngf** [comment]
One more point: Menu items aren't even sentences. All the examples we've seen so far apply to sentences. But yes, more opinions, please!

**Comment by janschreiber** [comment]
In German, an ellipsis has to be preceded by a space character, preferably a non-breaking-space, except when the omission is inside a word. No doubt about it, however common it may be to leave out the space in the GUI in other programs.
I'm not worried about the other changes by @dg9ngf though. Keeping the translation up to date has been more of a burden for me recently.
@dg9ngf, welcome on board. I hereby declare you the official German Notepad++ translator. One more thing off my chest. May the Force be with you. Feel free to contact me if you want advice.

**Comment by donho** [comment]
@dg9ngf Are you ready to take this responsability? :)

**Comment by Krzysiu** [comment]
Excuse me for talking here, as I don't know German. I already feel stupid. I don't want to learn you, I want to share my experiences in the moment of the big change and important discussion. I often translate webapps or standalone apps and I want to say a few things. Sorry if things I say are obvious. And sorry about my English. I'm EN>PL translator, not PL>EN for the sake of recipients ;) 

At first, please keep in mind that many translator does that, because they think that language knowledge is translating ability. That's of course false. That's why I don't think that other apps should be an example.

Also most translations are made by IT guys who are used to English grammar, English typography, English words etc. That makes their translation biased and americanized (Englishized? ;)). 

Language rules were mostly created long time ago. With world going faster and faster, language also is changing, so there's a chance that some construction is already allowed. That's why I recommend to check language rules every time there are some doubts. Relying on our knowledge is not always good because our teacher could be wrong, language rules may have some additional exception in digital world (sometimes exceptions allowed only in digital usage).

Do you have some "language support" in your country? In Poland most universities have webpages where you can ask question and experts will reply. Remember that some rules are already widely used (language usus), but not codified yet - like local forms of "Googling" are widely used worldwide, but they are normalized in just a few dictionaries, like English. It's also a nice thing to decide and answer clearly - "does my translation prefers usus or codex?".

My protip: experts could be conservative or liberal. If conservative expert will give you liberal answer (and vice-versa), then it's probably the best opinion :)

That's a bunch of my tips, but I want to get to my point. Ellipsis without space before it could be americanized version of German grammar. If experts say it's still wrong to ommit that space, then remember - if you'd ommit it then it will be reason for another translators to ignore space in their translations. You can end this loop by using correct German, not one of translator who was inspired by translators who got used to bad translations.

I think it could make you proud - most tools have bad translation, but look, I actually care about correctness! Even app like N++ can change language awareness of people and it could learn people what's the correct version. 

Also thanks for caring about such "small thing". Polish translators of many many apps doesn't seem to care about anything. 

Good luck, neighbors!

**Comment by h-h-h-h** [comment]
> @Krzysiu:
if you'd ommit it [the space] then it will be [a] reason for another translators to ignore space[s] in their translations. You can end this loop by using correct German, not one of [a] translator who was inspired by translators who got used to bad translations.
&vellip;
> Even app[s] like N++ can change [the] language awareness of people and it could learn [teach] people what's the correct version.

Thanks for your statement! I read it as an encouragement for decoupling a little bit from the praxis in Windows programs, and somehow as a pro-vote for spaces before ellipses.

**Comment by odysseuscm** [comment]
Hi, to make extra sure I double checked with the "Duden" (THE authority on German language): http://www.duden.de/sprachwissen/rechtschreibregeln/auslassungspunkte and of course it confirms what janschreiber wrote. If you omit a word there has to be a space before the ellipsis.
I guess the common use to not have that space has the simple (optical) reason that it's rather difficult to type a non-breaking-space and you don't want to take the chance that the ellipsis moves to the next line but rather have it "sticking" to the last word.
But I'm all for the correct use, so please leave the (non-breaking) spaces before the ellipses in.

**Comment by h-h-h-h** [comment]
> @odysseuscm:
I guess the common use to not have that space has the simple (optical) reason that it's rather difficult to type a non-breaking-space and you don't want to take the chance that the ellipsis moves to the next line but rather have it "sticking" to the last word.

Thanks for dropping in! Did you understand that the pull request is just about menu items? Notepad++ doesn't have menu items with line breaks.

**Comment by odysseuscm** [comment]
@h-h-h-h 
Yes, I did. That passage was just about the general problem people have with spaces before ellipses - ignore it, it wasn't really necessary ;-)
To make my opinion perfectly clear: The spaces before ellipses in the German menu items should remain! That's the correct way and it shouldn't bother us that Microsoft, Adobe etc. all do it wrong.

**Comment by dg9ngf** [comment]
Great, so now it's us against the world. I still don't like this new inconsistency. When I was advocating typography things in the past, there was at least some consent, but this time it's very little. To do things right, the three dots would need to be replaced by the ellipsis character as well. Let's see where we can improve typography in the UI elsewhere. How about the dash between file name and application title? But be aware that where " – " (en dash) is common in German, last time I checked at least US English seems to prefer "—" (em dash, no spaces). Not sure about world English now. Maybe some quotes and apostrophes could also be improved in the English translation as well. (Side note: The quotes are broken in the Tahoma font used before Vista, also in Verdana.)

I'd wish the same level of detail and correctness could be found in other parts of the UI. I'm thinking at the settings dialog that feels quite messy. It's even hard to translate that because there's so little space in it and it just won't extend for the content.

Well okay, you get your spaces back. I still don't agree with that. I'll update the pull request next week (I'm away for the weekend and getting more busy now).

Am I ready for this responsibility? I don't even know what my task would be. I was just fixing some things I found. I sure can do it in the future if something needs to be updated. I'm not watching the development progress very closely so somebody might need to notify me before a new release.

**Comment by h-h-h-h** [comment]
> @dg9ngf:
To do things right, the three dots would need to be replaced by the ellipsis character as well.

Yes, it's done in Audacity, too.

> How about the dash between file name and application title? But be aware that where " – " (en dash) is common in German, last time I checked at least US English seems to prefer "—" (em dash, no spaces).

For links to arguments and style recommendations see: https://github.com/icsharpcode/SharpDevelop/pull/707. This window title parts separator should be <q>translatable.</q> Would you even comment on this issue because there's opposition?

> Let's see where we can improve typography in the UI elsewhere.

See pull request https://github.com/notepad-plus-plus/notepad-plus-plus/pull/845#issuecomment-137253651.

> I'm thinking at the settings dialog that feels quite messy. It's even hard to translate that because there's so little space in it and it just won't extend for the content.

That's very bad!

**Comment by Krzysiu** [comment]
> Thanks for your statement! I read it as an encouragement for decoupling a little bit from the praxis in Windows programs, and somehow as a pro-vote for spaces before ellipses.

If that's correct, then yes! :) I understand very well why people don't use spaces before ellipses - it's just rare worldwide. But that's German and German rules are most important.

One tip more. It applies to Polish, but maybe you have something like that in Germany. It applies e.g. for hyphens, dashes and all things that I see just as "minus" ;) Rules about "minuses" are rather minor and experts allows to ommit them if there's a technical problem. So if you are writting in LibreOffice using straight "-" - it's wrong. When you do it on typewriter - it's not a language error in any way.

But both in Polish and English some "meta-rule" have a high priority. If there are technical difficulties in one part of the work, it's better to keep it the same in whole work. 

So _maybe_ there's something like that in German. I see lack of space in some parts of Notepad++ as a reason to ignore some minor rules and that wouldn't be an error. But if such step would be taken, maybe it would be better to ignore that rule in the whole tool.

Sorry, I won't give tips anymore, as I see that I have to many "maybe" and I shouldn't talk about things that I can say only by comparison with other languages :) I'm afraid it looks like "some Pole" teaches you about your language. If my tips would be useful... But I'm afraid I'm just taking your time. Tschuss Damen und Herren! :)

**Comment by h-h-h-h** [comment]
> @Krzysiu:
I'm afraid I'm just taking your time

To be honest, I didn't understand anything of your last post. Moreover, you write texts that are too long for my taste.   :frowning:

**Comment by Krzysiu** [comment]
> To be honest, I didn't understand anything of your last post
That would be my sleep deprivation.
tl;dr sometimes technical limits could force you to disobey language rules, yet it will be correct text, as it's "not your fault"; if you choose one version of rule or you decide to disobey it (like minus instead hyphen), you should stick to it through your whole project

> Moreover, you write texts that are too long for my taste
I think that's not a matter of taste - my long posts are my big disadvantage and I know that less people will read it because its length and problems with sticking to the topic, even my own. Sorry again!

**Comment by dg9ngf** [comment]
I've updated the file.

@xylographe "Count: X matches." does not seem to be localisable. It comes from the code itself. I can only edit the XML file.

**Comment by janschreiber** [comment]
I have been the main translator for German for several years now and I agree that @dg9ngf's commit is a great improvement. Please accept.

**Comment by donho** [comment]
https://github.com/notepad-plus-plus/notepad-plus-plus/commit/6b16ba2ef32e62397d1f17bb7c7e4f1b592f102a
